orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileManagerImpl.java

```diff
@@ -382,50 +382,63 @@ public OrcidProfile createOrcidProfileAndNotify(OrcidProfile orcidProfile) {
         return createdOrcidProfile;
     }
 
-    @Override
-    @Transactional
+    @Override    
     public OrcidProfile updateOrcidProfile(OrcidProfile orcidProfile) {
         return updateOrcidProfile(orcidProfile, UpdateOptions.ALL);
     }
 
-    @Override
-    @Transactional
-    public OrcidProfile updateOrcidProfile(OrcidProfile orcidProfile, UpdateOptions updateOptions) {
+    private OrcidProfile updateOrcidProfile(OrcidProfile orcidProfile, UpdateOptions updateOptions) {
         String amenderOrcid = sourceManager.retrieveSourceOrcid();
-        ProfileEntity existingProfileEntity = profileDao.find(orcidProfile.getOrcidIdentifier().getPath());
-
-        Visibility defaultVisibility = Visibility.fromValue(existingProfileEntity.getActivitiesVisibilityDefault().value());
-
-        if (existingProfileEntity != null) {
-            // Dont delete the existing elements anymore
-            // profileDao.removeChildrenWithGeneratedIds(existingProfileEntity);
-            setWorkPrivacy(orcidProfile, defaultVisibility);
-            setAffiliationPrivacy(orcidProfile, defaultVisibility);
-            setFundingPrivacy(orcidProfile, defaultVisibility);
-        }
-        dedupeWorks(orcidProfile);
-        dedupeAffiliations(orcidProfile);
-        dedupeFundings(orcidProfile);
-        addSourceToEmails(orcidProfile, existingProfileEntity, amenderOrcid);
-
-        Boolean claimed = orcidProfile.getOrcidHistory() != null ? orcidProfile.getOrcidHistory().isClaimed() : existingProfileEntity.getClaimed();
-
-        if (orcidProfile.getOrcidInternal() != null && orcidProfile.getOrcidInternal().getPreferences() != null
-                && orcidProfile.getOrcidInternal().getPreferences().getActivitiesVisibilityDefault() != null) {
-            defaultVisibility = orcidProfile.getOrcidInternal().getPreferences().getActivitiesVisibilityDefault().getValue();
-        }
-        addDefaultVisibilityToBioItems(orcidProfile, defaultVisibility, claimed);
+        transactionTemplate.execute(new TransactionCallbackWithoutResult() {
```

Comment:
Hi @wjrsimpson,

So here is where magic happens, instead of having a _@transactional_ on each method that invoke _updateOrcidProfile(OrcidProfile orcidProfile, UpdateOptions updateOptions)_, I removed all transactional annotations from them and execute this method into a single transaction.

Thanks